--- a/hdf5/configure
+++ b/hdf5/configure
@@ -7925,7 +7925,8 @@ if ac_fn_fc_try_run "$LINENO"; then :


         pac_validIntKinds="`sed -n '1p' pac_fconftest.out`"
-	pac_validRealKinds="`sed -n '2p' pac_fconftest.out`"
+	#pac_validRealKinds="`sed -n '2p' pac_fconftest.out`"
+	pac_validRealKinds="1,2,3"
         PAC_FC_MAX_REAL_PRECISION="`sed -n '3p' pac_fconftest.out`"

 cat >>confdefs.h <<_ACEOF
@@ -8097,7 +8098,8 @@ if ac_fn_fc_try_run "$LINENO"; then :


         pac_validIntKinds="`sed -n '1p' pac_fconftest.out`"
-	pac_validRealKinds="`sed -n '2p' pac_fconftest.out`"
+	#pac_validRealKinds="`sed -n '2p' pac_fconftest.out`"
+	pac_validRealKinds="1,2,3"
         PAC_FC_MAX_REAL_PRECISION="`sed -n '3p' pac_fconftest.out`"

 cat >>confdefs.h <<_ACEOF

