--- a/esmf/build_config/Darwin.nag.default/build_rules.mk
+++ b/esmf/build_config/Darwin.nag.default/build_rules.mk
@@ -115,6 +115,12 @@ ESMF_CLANGSTR := $(findstring clang, $(shell $(ESMF_CXXCOMPILER) --version))
 # (e.g., ABORT) are made.
 ESMF_F90COMPILEOPTS += -DESMF_NAG_UNIX_MODULE

+############################################################
+# Currently NAG does not support the Fortran2018 assumed type feature
+#
+ESMF_F90COMPILECPPFLAGS += -DESMF_NO_F2018ASSUMEDTYPE
+ESMF_CXXCOMPILECPPFLAGS += -DESMF_NO_F2018ASSUMEDTYPE
+
 ############################################################
 # nag currently does not support OpenMP
 #

