diff --git before/esmf/build_config/Linux.nvhpc.default/build_rules.mk after/esmf/build_config/Linux.nvhpc.default/build_rules.mk
--- before/esmf/build_config/Linux.nvhpc.default/build_rules.mk
+++ after/esmf/build_config/Linux.nvhpc.default/build_rules.mk
@@ -121,10 +121,6 @@ ESMF_CXXCOMPILER_VERSION    = ${ESMF_CXXCOMPILER} --version -c
 ESMF_CCOMPILER_VERSION      = ${ESMF_CCOMPILER} --version -c
 
 ############################################################
-# Currently no support the Fortran2018 assumed type feature
-#
-ESMF_F90COMPILECPPFLAGS += -DESMF_NO_F2018ASSUMEDTYPE
-ESMF_CXXCOMPILECPPFLAGS += -DESMF_NO_F2018ASSUMEDTYPE
 
 ############################################################
 # Construct the ABISTRING
