--- szlib/src/rice.c	2017-01-25 13:06:17
+++ szlib/src/rice.c	2025-04-24 11:46:06
@@ -103,8 +103,8 @@
 #endif /* !HDF */

 #ifndef REMOVE_SZIP_ENCODER
-static int (*winner_function)();
-static int (*winner_ref_function)();
+static int (*winner_function)(unsigned *, unsigned *);
+static int (*winner_ref_function)(unsigned *, unsigned *);
 #endif /* !defined(REMOVE_SZIP_ENCODER) */

 static boolean compress_exactly_as_chip;
