--- /dev/null
+++ b/FMS/cmake/compiler_flags_LLVMFlang_Fortran.cmake
@@ -0,0 +1,13 @@
+# Precision-based Fortran compiler flags
+set(r8_flags "-fdefault-real-8 -fdefault-double-8") # Fortran flags for 64BIT precision
+set(r4_flags "-fdefault-real-4") # Fortran flags for 32BIT precision
+
+# LLVM Flang Fortran
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}")
+
+set(CMAKE_Fortran_FLAGS_RELEASE "-O2")
+set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -g")
+
+set(CMAKE_Fortran_LINK_FLAGS "" )
+
+#ufs flags -- Not known
--- a/FMS/cmake/fms_compiler_flags.cmake
+++ b/FMS/cmake/fms_compiler_flags.cmake
@@ -8,6 +8,8 @@ elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
   include(compiler_flags_Intel_Fortran)
 elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "IntelLLVM")
   include(compiler_flags_IntelLLVM_Fortran)
+elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "LLVMFlang")
+  include(compiler_flags_LLVMFlang_Fortran)
 else()
   message(WARNING "Fortran compiler with ID ${CMAKE_Fortran_COMPILER_ID} will be used with CMake default options")
 endif()
