diff --git before/nccmp/src/getopt.h after/nccmp/src/getopt.h
index 18e1026..c3193e7 100644
--- before/nccmp/src/getopt.h
+++ after/nccmp/src/getopt.h
@@ -143,7 +143,7 @@ struct option
    errors, only prototype getopt for the GNU C library.  */
 extern int getopt (int __argc, char *const *__argv, const char *__shortopts);
 # else /* not __GNU_LIBRARY__ */
-extern int getopt ();
+extern int getopt (int __argc, char *const *__argv, const char *__shortopts);
 # endif /* __GNU_LIBRARY__ */

 # ifndef __need_getopt
